xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -84,41 +84,76 @@ public ManagementServerNotFoundException(String msg) {
   }
 
   /**
-   * Factory method for creating a {@link XdsClientWrapperForServerSds}.
+   * Creates a {@link XdsClientWrapperForServerSds}.
    *
    * @param port server's port for which listener config is needed.
-   * @param bootstrapper {@link Bootstrapper} instance to load bootstrap config.
-   * @param syncContext {@link SynchronizationContext} needed by {@link XdsClient}.
    */
-  public static XdsClientWrapperForServerSds newInstance(
-      int port, Bootstrapper bootstrapper, SynchronizationContext syncContext)
-      throws IOException, ManagementServerNotFoundException {
-    Bootstrapper.BootstrapInfo bootstrapInfo = bootstrapper.readBootstrap();
-    final List<Bootstrapper.ServerInfo> serverList = bootstrapInfo.getServers();
-    if (serverList.isEmpty()) {
-      throw new ManagementServerNotFoundException("No management server provided by bootstrap");
+  public XdsClientWrapperForServerSds(int port) {
+    this.port = port;
+  }
+
+  private SynchronizationContext createSynchronizationContext(int port) {
```

Comment:
you can just use instance variable `port`. 

```suggestion
  private SynchronizationContext createSynchronizationContext() {
```